DASH-MPD-PATCH-RESTRICTED.xsd

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE schema [
+ <!ENTITY ncname "\i\c*">
+ <!ENTITY qname  "(&ncname;:)?&ncname;">
+ <!ENTITY aname  "@&qname;">
+ <!ENTITY pos    "\[\d+\]">
+ <!ENTITY attr   "\[&aname;='(.)*'\]|\[&aname;=&quot;(.)*&quot;\]">
+ <!ENTITY cond   "&attr;|&pos;">
+ <!ENTITY step   "(&qname;|\*)(&cond;)*">
+ <!ENTITY nspa   "namespace::&ncname;">
+ <!ENTITY last   "(&step;|&aname;|&nspa;)">
+]>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:patch="urn:ietf:params:xml:schema:patch-ops" xmlns="urn:mpeg:dash:schema:mpd-patch:2020" targetNamespace="urn:mpeg:dash:schema:mpd-patch:2020" elementFormDefault="qualified" attributeFormDefault="unqualified">
+    <!-- Import of patch operations from RFC5261 -->
+    <xs:import namespace="urn:ietf:params:xml:schema:patch-ops" schemaLocation="https://www.iana.org/assignments/xml-registry/schema/patch-ops.xsd"/>
+    <xs:element name="Patch" type="PatchType"/>
+    <!-- Patch -->
+    <xs:complexType name="PatchType">
+        <xs:choice minOccurs="0" maxOccurs="unbounded">
+            <xs:element name="add" type="add-restricted"/>
+            <xs:element name="remove" type="remove-restricted"/>
+            <xs:element name="replace" type="replace-restricted"/>
+            <xs:any namespace="##other" processContents="lax"/>
+        </xs:choice>
+        <xs:attribute name="mpdId" type="xs:string"/>
+        <xs:attribute name="publishTime" type="xs:dateTime"/>
+        <xs:attribute name="originalPublishTime" type="xs:dateTime"/>
+        <xs:anyAttribute namespace="##other" processContents="lax"/>
+    </xs:complexType>
+    <!-- Restricted XPath Selector for Replace/Remove Elements -->
+    <xs:simpleType name="xpath-restricted">
+        <xs:restriction base="patch:xpath">
+            <xs:pattern value="(/)?(&step;/)*(&last;)"/>
```

Comment:
Simplified from [`(/)?((&id;)((/&step;)*(/&last;))?|(&step;/)*(&last;))`](https://regexper.com/#%28%5C%2F%29%3F%28%28%26id%3B%29%28%28%5C%2F%26step%3B%29*%28%5C%2F%26last%3B%29%29%3F%7C%28%26step%3B%5C%2F%29*%28%26last%3B%29%29)

Namely the `&id;` matcher was removed which eliminates the alternate pathways. The match remains:

- A possible root node indicator
- Followed by zero or more steps, where step is a node with optional attribute or positional qualifiers
- Finished by either a step, an attribute, or a namespace selector

Potential question: Do we require all paths to be built from root and remove the optionality to make the patch more definite? This would align more with the intention of the restrictions in my opinion.